index.js

```diff
@@ -114,6 +118,9 @@ module.exports = function requireDir(dir, opts) {
                     }
                 } else {
                     map[base] = require(path);
+                    if (camelcase) {
+                        map[toCamelcase(base)] = map[base];
+                    }
```

Comment:
Your implementation looks good, but just wondering, might it be better if we did this aliasing in one spot just below `return map`, rather than inline separately for directories vs. files?

```js
if (opts.camelcase) {
    for (var base of map) {
        // protect against enumerable object prototype extensions:
        if (!map.hasOwnProperty(base)) {
            continue;
        }

        map[toCamelCase(base)] = map[base];
    }
}
```

This would still cover the `recurse` case.